reusables/default_list.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+class defaultlist(list):
+    def __init__(self, *args, factory=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.factory = factory
+
+    def __getitem__(self, index):
+        if index >= len(self):
+            diff = index - len(self) + 1
+            for i in range(diff):
+                self.append(self.factory())
```

Comment:
I had forgotten about that ahah. Now that I see it with fresh eyes, it would have been better to use list multiplication to achieve the same effect (as it's much faster than append).

Instead of 

```python
for i in range(diff):
    self.append(self.factory())
```

we could do 
```python
self.extend([self.factory()] * diff)
```

That second method has a disadvantage though, which is that if the factory is random.random for instance (or any other non deterministic callable), well, the random value will be the same for all the intermediate items in the list.

Reply:
While faster, I do think that could have unexpected behavior. For example if you wanted to have a counter or something: 

```python
idx = 0 
def counter():
     global idx
     idx += 1 
     return idx

a = defaultlist(factory=counter)
a[1]
# 2
```

Reply:
There's also the list comprehension way that would be faster than the append way, and would avoid this unwanted behaviour.

```python
self.extend([self.factory() for _ in range(diff)])
```

Reply:
Oh good call! Actually going to make it a generator for lazy eval: 

```python
    def __getitem__(self, index):
        if index >= len(self):
            self.extend(self.factory() for _ in range(index - len(self) + 1))
        return super().__getitem__(index)
```